api/src/test/java/com/epam/pipeline/controller/pipeline/PipelineControllerTest.java

```diff
@@ -600,12 +589,11 @@ public void shouldUploadFile() throws Exception {
         final MvcResult mvcResult = performRequest(post(String.format(PIPELINE_ID_FILE_UPLOAD_URL, ID))
                         .params(multiValueMapOf(PATH, TEST_STRING)).content(MULTIPART_CONTENT),
                 MULTIPART_CONTENT_TYPE, EXPECTED_CONTENT_TYPE);
+        final ArgumentCaptor<UploadFileMetadata> captor = ArgumentCaptor.forClass(UploadFileMetadata.class);
 
-        verify(mockPipelineApiService).uploadFiles(eq(ID), eq(TEST_STRING), anyListOf(UploadFileMetadata.class));
-        final String expectedResult = mvcResult.getResponse().getContentAsString();
-        assertThat(expectedResult).contains(MULTIPART_CONTENT_FILE_NAME)
-                .contains(FILE_SIZE)
-                .contains(MediaType.APPLICATION_OCTET_STREAM_VALUE);
+        verify(mockPipelineApiService).uploadFiles(eq(ID), eq(TEST_STRING),
+                Collections.singletonList(captor.capture()));
+        assertResponse(mvcResult, "[{fileName=file.txt, fileSize=0 Kb, fileType=application/octet-stream}]");
```

Comment:
Could you please initiate a new instance of expected object? Something like the following:
```java
UploadFileMetadata expectedUploadFileMetadata = new ...;
assertResponse(mvcResult, Collections.singletonList(expectedUploadFileMetadata));
```

Reply:
In this case we can also get rid of new `assertResponse(MvcResult, String)` method.